### PR TITLE
HTTP responses

### PR DESCRIPTION
**Pull request by Hermesiss**
You can review existing code while we are waiting for https://github.com/wakatime/wakatime/issues/181

### CODE REVIEW

File: Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
-#if (UNITY_EDITOR)
```
Comment (vladfaust) [inline Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs]: It should be on top, for beauty 🍃

File: Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs [hunk 2]
```diff
@@ -131,7 +133,8 @@ struct Heartbeat {
         : string.Empty;
 
       var heartbeat = new Heartbeat(file, fromSave);
-      if ((heartbeat.time - _lastHeartbeat.time < HEARTBEAT_COOLDOWN) && !fromSave &&
+      if ((heartbeat.time - _lastHeartbeat.time < HEARTBEAT_COOLDOWN) && !fromSave && 
+        //BUG: _lastHeartbeat has no time, entity or type
```
Comment (vladfaust) [inline Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs]: It's a good practice to not intervent long conditionals (even multi-line) with comments, put it on top instead.

File: Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs [hunk 3]
```diff
@@ -146,14 +149,16 @@ struct Heartbeat {
 
       request.SendWebRequest().completed +=
         operation => {
-          if (request.downloadHandler.text == string.Empty) {
+          /*if (request.downloadHandler.text == string.Empty) {
```
Comment (vladfaust) [inline Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs]: Should put a comment on top of the block explaining why is it commented.

File: Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs [hunk 4]
```diff
@@ -175,6 +180,41 @@ struct Heartbeat {
         };
     }
 
+    /// <summary>
+    /// Checks for status code and takes correspond actions
```
Comment (vladfaust) [inline Assets/com.vladfaust.unitywakatime/Editor/Plugin.cs]:
```suggestion
    /// Checks for status code and then acts accordingly
```

**Comment by vladfaust** [comment]
I'm sorry for the delay, @Hermesiss, I'm a bit busy with messy containers security! 

First of all, thanks for your time. I do appreciate your help on the project.

I guess this PR is aimed to suppress malformed Wakatime responses. That's good, I think, as it's a third-party bug and we don't want to disrupt the user experience.

A small note — it's generally bad practice to include commented out blocks of code into the master branch. We always have the history, so we could add it back once it's working as intended.

I think the `HandleHttpResponseSuccess` method is too burden, isn't it enough to check `response.error != null` and `response.status` in `(200..300)` range? And if it's out of that range, just print `Got 4xx error`...

**Comment by vladfaust** [comment]
I may be in some way out of understanding the real purpose of this PR. Which issue particularly does it fix? As per #13, the problem was malformed response, which is now solved. 

In everyday programming (especially for such small projects as this) we should rely on third party API to work as expected most of the time, handling all errors in bulk with minimum code. This reduces the complexity and therefore improves maintainability.

In this case, expected error is 429 only, all other are unexpected and should throw. There is no need in this `HandleHttpResponseSuccess ` method, in my opinion.

**Comment by vladfaust** [comment]
Additionally, for better communication, you should split the PRs as much as possible, as it's sometimes hard to follow their purpose. Single PR = single change (or complex overhaul).

**Comment by Hermesiss** [comment]
There are 3 important for us error codes that require special reactions: `0`, `429` and `500`. 

While working on `429` I tried to temporally increase `HEARTBEAT_COOLDOWN`. I found out that the part where it's used is broken due to some undocumented API changes.

In this PR I want to move scattered data checks to one place. I took all possible codes from [dev section](https://wakatime.com/developers).
